linuxPackages.rtl8852au: fix build

eljamm opened the pull request:
for kernels >= 6.13 and 6.14


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

eljamm committed:
linuxPackages.rtl8852au: fix build

for kernels >= 6.13 and 6.14

Aleksanaa commented on pkgs/os-specific/linux/rtl8852au/default.nix:
```diff
@@ -27,6 +28,14 @@ stdenv.mkDerivation (finalAttrs: {
+  patches = [
+    # fix builds for kernels >= 6.13 and 6.14
+    (fetchpatch2 {
+      url = "https://patch-diff.githubusercontent.com/raw/lwfinger/rtl8852au/pull/115.patch";
```
Please vendor the patch. See https://github.com/NixOS/nixpkgs/tree/master/pkgs#patches

eljamm replied:
Will do, thanks

lonyelon reviewed (approved):
Nice, thanks for taking care of this package, I was away and have no 8852au devices anymore, so I didn't notice the break.

As for the patch, it is good, but maybe it should be sent as a PR for the upstream repo. The problem is that, sadly, Larry W. Finger (the developer/maintainer for these drivers) passed away last year, and as far as I know no one has stepped up to replace him.

Thanks for the PR and rest in peace Larry, you really helped the community.

eljamm commented:
> As for the patch, it is good, but maybe it should be sent as a PR for the upstream repo.

The patch actually came [from the upstream repo](https://github.com/lwfinger/rtl8852au/pull/115), it was just vendored because the PR is still open and therefore the patch might change. do you think I should add a comment as a reference or is it fine like this?

> The problem is that, sadly, Larry W. Finger (the developer/maintainer for these drivers) passed away last year, and as far as I know no one has stepped up to replace him.

Yeah, I've seen someone mention this, upstream. Hope he rests in peace.

eljamm commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 410672`

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/15232258365


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8852au</li>
    <li>linuxPackages_5_4_hardened.rtl8852au</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_5_10_hardened.rtl8852au (linuxKernel.packages.linux_5_10_hardened.rtl8852au)</li>
    <li>linuxPackages_5_15_hardened.rtl8852au (linuxKernel.packages.linux_5_15_hardened.rtl8852au)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 20 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.rtl8852au</li>
    <li>linuxKernel.packages.linux_5_15.rtl8852au</li>
    <li>linuxKernel.packages.linux_5_4.rtl8852au</li>
    <li>linuxKernel.packages.linux_6_1.rtl8852au</li>
    <li>linuxPackages.rtl8852au (linuxKernel.packages.linux_6_12.rtl8852au)</li>
    <li>linuxPackages_hardened.rtl8852au (linuxPackages_6_12_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_6_13.rtl8852au</li>
    <li>linuxPackages_6_13_hardened.rtl8852au (linuxKernel.packages.linux_6_13_hardened.rtl8852au)</li>
    <li>linuxPackages_latest.rtl8852au (linuxKernel.packages.linux_6_14.rtl8852au)</li>
    <li>linuxPackages_6_14_hardened.rtl8852au (linuxKernel.packages.linux_6_14_hardened.rtl8852au)</li>
    <li>linuxPackages_6_1_hardened.rtl8852au (linuxKernel.packages.linux_6_1_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_6_6.rtl8852au</li>
    <li>linuxPackages_6_6_hardened.rtl8852au (linuxKernel.packages.linux_6_6_hardened.rtl8852au)</li>
    <li>linuxPackages_ham.rtl8852au (linuxKernel.packages.linux_ham.rtl8852au)</li>
    <li>linuxPackages_latest-libre.rtl8852au (linuxKernel.packages.linux_latest_libre.rtl8852au)</li>
    <li>linuxPackages-libre.rtl8852au (linuxKernel.packages.linux_libre.rtl8852au)</li>
    <li>linuxPackages_lqx.rtl8852au (linuxKernel.packages.linux_lqx.rtl8852au)</li>
    <li>linuxPackages_xanmod.rtl8852au (linuxKernel.packages.linux_xanmod.rtl8852au)</li>
    <li>linuxPackages_xanmod_latest.rtl8852au (linuxKernel.packages.linux_xanmod_latest.rtl8852au, linuxPackages_xanmod_stable.rtl8852au)</li>
    <li>linuxPackages_zen.rtl8852au (linuxKernel.packages.linux_zen.rtl8852au)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>linuxPackages_5_10_hardened.rtl8852au</summary>
<pre>  343 |         struct phl_bk_module_ops bk_ops = {0};
      |                                            ^
/build/source/phl/phl_cmd_btc.c:343:44: note: (near initialization for 'bk_ops')
/build/source/phl/phl_cmd_btc.c:343:44: error: invalid initializer
/build/source/phl/phl_cmd_btc.c:343:44: note: (near initialization for 'bk_ops.<anonymous>')
cc1: some warnings being treated as errors
make[3]: *** [/nix/store/dxx0jy914w5l1zs41drm7a32g7pynynj-linux-hardened-5.10.237-dev/lib/modules/5.10.237-hardened1/source/scripts/Makefile.build:286: /build/source/phl/phl_cmd_btc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
/build/source/phl/phl_cmd_ps.c: In function 'phl_register_ps_module':
/build/source/phl/phl_cmd_ps.c:1504:44: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init-Werror=designated-init8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init%07-Werror=designated-init8;;%07])
 1504 |         struct phl_bk_module_ops bk_ops = {0};
      |                                            ^
/build/source/phl/phl_cmd_ps.c:1504:44: note: (near initialization for 'bk_ops')
/build/source/phl/phl_cmd_ps.c:1504:44: error: invalid initializer
/build/source/phl/phl_cmd_ps.c:1504:44: note: (near initialization for 'bk_ops.<anonymous>')
cc1: some warnings being treated as errors
make[3]: *** [/nix/store/dxx0jy914w5l1zs41drm7a32g7pynynj-linux-hardened-5.10.237-dev/lib/modules/5.10.237-hardened1/source/scripts/Makefile.build:286: /build/source/phl/phl_cmd_ps.o] Error 1
make[2]: *** [/nix/store/dxx0jy914w5l1zs41drm7a32g7pynynj-linux-hardened-5.10.237-dev/lib/modules/5.10.237-hardened1/source/Makefile:1840: /build/source] Error 2
make[1]: *** [/nix/store/dxx0jy914w5l1zs41drm7a32g7pynynj-linux-hardened-5.10.237-dev/lib/modules/5.10.237-hardened1/source/Makefile:192: __sub-make] Error 2
make: *** [Makefile:639: modules] Error 2</pre>
</details>
<details>
<summary>linuxPackages_5_15_hardened.rtl8852au</summary>
<pre>/build/source/phl/phl_cmd_btc.c:343:44: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init-Werror=designated-init8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init%07-Werror=designated-init8;;%07])
  343 |         struct phl_bk_module_ops bk_ops = {0};
      |                                            ^
/build/source/phl/phl_cmd_btc.c:343:44: note: (near initialization for 'bk_ops')
/build/source/phl/phl_cmd_btc.c:343:44: error: invalid initializer
/build/source/phl/phl_cmd_btc.c:343:44: note: (near initialization for 'bk_ops.<anonymous>')
cc1: some warnings being treated as errors
make[2]: *** [/nix/store/vf1hpv6yca3118rcc2crm3rxkpadj7z4-linux-hardened-5.15.182-dev/lib/modules/5.15.182-hardened1/source/scripts/Makefile.build:289: /build/source/phl/phl_cmd_btc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/build/source/phl/phl_cmd_ps.c: In function 'phl_register_ps_module':
/build/source/phl/phl_cmd_ps.c:1504:44: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [8;;[https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init-Werror=designated-init8;;]](https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdesignated-init%07-Werror=designated-init8;;%07])
 1504 |         struct phl_bk_module_ops bk_ops = {0};
      |                                            ^
/build/source/phl/phl_cmd_ps.c:1504:44: note: (near initialization for 'bk_ops')
/build/source/phl/phl_cmd_ps.c:1504:44: error: invalid initializer
/build/source/phl/phl_cmd_ps.c:1504:44: note: (near initialization for 'bk_ops.<anonymous>')
cc1: some warnings being treated as errors
make[2]: *** [/nix/store/vf1hpv6yca3118rcc2crm3rxkpadj7z4-linux-hardened-5.15.182-dev/lib/modules/5.15.182-hardened1/source/scripts/Makefile.build:289: /build/source/phl/phl_cmd_ps.o] Error 1
make[1]: *** [/nix/store/vf1hpv6yca3118rcc2crm3rxkpadj7z4-linux-hardened-5.15.182-dev/lib/modules/5.15.182-hardened1/source/Makefile:1914: /build/source] Error 2
make: *** [Makefile:639: modules] Error 2</pre>
</details>
</details>


---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 12 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8852au</li>
    <li>linuxKernel.packages.linux_lqx.rtl8852au</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8852au</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8852au</li>
    <li>linuxKernel.packages.linux_xanmod_stable.rtl8852au</li>
    <li>linuxKernel.packages.linux_zen.rtl8852au</li>
    <li>linuxPackages_5_4_hardened.rtl8852au</li>
    <li>linuxPackages_lqx.rtl8852au</li>
    <li>linuxPackages_xanmod.rtl8852au</li>
    <li>linuxPackages_xanmod_latest.rtl8852au</li>
    <li>linuxPackages_xanmod_stable.rtl8852au</li>
    <li>linuxPackages_zen.rtl8852au</li>
  </ul>
</details>
<details>
  <summary>:x: 18 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.rtl8852au</li>
    <li>linuxPackages_5_10_hardened.rtl8852au (linuxKernel.packages.linux_5_10_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_5_15.rtl8852au</li>
    <li>linuxPackages_5_15_hardened.rtl8852au (linuxKernel.packages.linux_5_15_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_5_4.rtl8852au</li>
    <li>linuxKernel.packages.linux_6_1.rtl8852au</li>
    <li>linuxPackages.rtl8852au (linuxKernel.packages.linux_6_12.rtl8852au)</li>
    <li>linuxPackages_hardened.rtl8852au (linuxPackages_6_12_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_6_13.rtl8852au</li>
    <li>linuxPackages_6_13_hardened.rtl8852au (linuxKernel.packages.linux_6_13_hardened.rtl8852au)</li>
    <li>linuxPackages_latest.rtl8852au (linuxKernel.packages.linux_6_14.rtl8852au)</li>
    <li>linuxPackages_6_14_hardened.rtl8852au (linuxKernel.packages.linux_6_14_hardened.rtl8852au)</li>
    <li>linuxPackages_6_1_hardened.rtl8852au (linuxKernel.packages.linux_6_1_hardened.rtl8852au)</li>
    <li>linuxKernel.packages.linux_6_6.rtl8852au</li>
    <li>linuxPackages_6_6_hardened.rtl8852au (linuxKernel.packages.linux_6_6_hardened.rtl8852au)</li>
    <li>linuxPackages_ham.rtl8852au (linuxKernel.packages.linux_ham.rtl8852au)</li>
    <li>linuxPackages_latest-libre.rtl8852au (linuxKernel.packages.linux_latest_libre.rtl8852au)</li>
    <li>linuxPackages-libre.rtl8852au (linuxKernel.packages.linux_libre.rtl8852au)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>linuxKernel.packages.linux_5_10.rtl8852au</summary>
<pre>Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/15n0gdj62y8qz7gnwgsgx1yqx71hl1m3-115.patch
patching file os_dep/linux/ioctl_cfg80211.c
patching file os_dep/linux/ioctl_cfg80211.c
patching file os_dep/osdep_service_linux.c
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j4 SHELL=/nix/store/s1qkj0ph0ma64a6743mvkwnabrbw1hsc-bash-5.2p37/bin/bash ARCH=arm64 CONFIG_PLATFORM_I386_PC=n CONFIG_PLATFORM_ARM_RPI=y
#rm -f .symvers.8852au
make ARCH=arm64 CROSS_COMPILE= -C  M=/build/source  modules
make[1]: Entering directory '/build/source'
make[1]: *** M=/build/source: No such file or directory.  Stop.
make[1]: Leaving directory '/build/source'
make: *** [Makefile:639: modules] Error 2</pre>
</details>
</details>

eljamm committed:
linuxPackages.rtl8852au: only build for x86_64

as it appears that it doesn't work on aarch64. Also, mark as broken on
hardened kernels which are older than 6.*
